### ISSUE TITLE
Dell G15 5515 Ryzen Edition Fan Control Not Working

### ISSUE THREAD

**Issue by Killerpac** [T-Troll/alienfx-tools#127]
**Describe the bug**
An Error Message is thrown That it's not supported.

**Expected behavior**
Should Work As I Used an Old Version Before But After Bios Update it Broke

**Desktop (please complete the following information):**
 - Dell G15 5515 Ryzen Edition 
 - Windows 10 21H2

**Additional context**
Should I Provide A ACPI Dump

**Comment by Killerpac** [comment]
Ok Fixed Apparently Disabling Memory Integrity In Security Solved the problem But the Fans Are Not Spinning

**Comment by vinikmv** [comment]
Sorry if I'm getting in the way, but just to add that I have:
- Windows 10 21H2
-  Dell G15 5515 Ryzen Edition (R7/3060) with BIOS 1.6.0
- Last version of AlienFx-Tools (5.6.0.1)

And its working fine, but I updated the bios with the file available at the Dell drivers page. 
Updating the bios with Windows automatic updates only gave me headaches.

**Comment by T-Troll** [comment]
Thanks for info.
But anyway, it's interesting to have a dump from new BIOS to support it.

Close the bug if this fix it, i mention it into manual.

**Comment by T-Troll** [comment]
> Ok Fixed Apparently Disabling Memory Integrity In Security Solved the problem But the Fans Are Not Spinning

Fans are spin by BIOS, i only add boost to it. It's not direct, but indirect control!

**Comment by Pedro-Thales** [comment]
Hello I've the same notebook and I'm able to control the fans, but when the system is idle, the monitor turns off(after 10 minutes because of the power plan) and is locked by windows and requires a password to unlock, when I unlock it the cpu fan stay at 500 rpm all the time and the gpu stay at 0, not returning until I reboot.

I have:
- Windows 11 21H2
- Dell G15 5515 Ryzen Edition (R7/3060) with BIOS 1.6.0
- AlienFx-Control 5.9.4.3

**Comment by T-Troll** [comment]
Seems like thermal mode switched after lock.
Can you try not to reboot, but restart my app?

**Comment by Pedro-Thales** [comment]
I've tried before, with no success, but I think that maybe hwmonitor is interfering with it. I'll do some more tests with it

**Comment by Pedro-Thales** [comment]
Is not working, if I try to restart this app after being the fan being stuck with some value, not always is 500, don`t work the app does not open and I can see two process of aliexfx-gui.exe in the task manager, but it does not open, and after a few seconds I have driver state power failure BSOD on windows 11

**Comment by T-Troll** [comment]
>  I can see two process of aliexfx-gui.exe 

Wow! Seems like it's a bug in Windows Schedule OR `alienfx-gui` does not exit correctly.
Can you first try to disable "start with windows" and start it manually?
Is it will be two of it after lock/unlock?
(In fact, having 2 copy doesn't make a lot of troubles, but can provide a bit of mess in config if you are using keyboard shortcut).

> maybe HWInfo64 is interfering

Hmm... Simple to check - disable fan control into HWInfo (keep it "automatic).

**Comment by Pedro-Thales** [comment]
![image](https://user-images.githubusercontent.com/8193588/171967721-08c723a0-04ff-453e-b21e-b8a267d4f3fe.png)
I've tried everything you said but it gets stuck sometimes this time was while playing, and it is stuck I've tried restarting the app several times, this time it did not have a stuck process but the fans stay with 400 rpm even though in the graph says that it should be at almost 100% 

A solution to get it unstuck was to close it and then open hwinfo and control the fans from there, then set to autom close hwinfo and then open alienfx, but when I open it, it loses the overboost values as you can see in the image below, it should be at ~5400 but it is at 4400 rpm
![image](https://user-images.githubusercontent.com/8193588/171968146-3ca95817-5e3a-4238-aa23-f7988d43af06.png)

**Comment by T-Troll** [comment]
Thank you for pictures.
BTW, you can use `alienfan-gui` if you don't need light control - it's more light.

So, what i see... It's really hello wave from HWINFO. And THAT'S why i don't use the same approach.
They limit fan RPMs via EC, so the fan can't run faster in any case.
My app report what it sees from hardware, so it reveals real situation.

Some words about overboost. First, check it carefully - so much overboost can stop the fan! I check this situation at detection, but sometimes my app can miss the point. So good idea to check it again.
Also, at latest versions of app it is some seconds delay enabling it - because at some gear fans spin up slow then overboosted. So first it set 100% (without overboost), wait some seconds, then spin more. It's way faster.

PS: I recommend to spin GPU fan earlier - for Nvidia throttling start at 86C,  so better keep it lower. 90C is CRT temp, can be serious performance drop or even black screen/BSOD.

**Comment by Pedro-Thales** [comment]
Ok let me try that. 

Is there a way to do edit manually the overboost? I'm asking because as you said the overboost may miss the point and I know when I used the AWCC when pressing the G key in the key board it stays at 5400 rpm and it would be good to set it to something less than that so that the noise is not that high and the thermals are better than 4200 that its "normal" 100%

**Comment by T-Troll** [comment]
> Is there a way to do edit manually the overboost?

Yes, even 2 ways.
1. (easy) - `alienfan-cli setover=<fan>,<boost>`
2. (harder) - stop gui, edit registry (HKCU/SOFTWARE/Alienfan/Boost-<fan id>) - first byte is boost, next 2 bytes - max. RPMs

**Comment by Pedro-Thales** [comment]
So I just enter for example: `alienfan-cli setover=0,5200` for a boost maximum of 5200 rpm? 

I knew it but I must confess that I don't really know what they are for :(  

This power modes are defined in the alienfx-gui? Can I define different profiles for the fans in each power mode? What I can change from one  power mode to another

**Comment by T-Troll** [comment]
No. Here is a boost value, so 0,100 1,125 or so. CLI test it and define RPMs by itself.

No, power modes defined in BIOS it also alter PLs and some other stuff. Yes, you need to use profiles from gui to switch it (but, being honest, rare stuff to do so - you don't win anything re-defining it).

**Comment by Pedro-Thales** [comment]
I tried it but with no success
![image](https://user-images.githubusercontent.com/8193588/172006186-596724d1-c20a-496f-a49b-90b82393a3d8.png)

I've tried to set as you said but the result was the same it was stuck on 1300rpm
![image](https://user-images.githubusercontent.com/8193588/172006383-f34f26a3-c065-4444-8854-4c7de2d8e1b8.png)

The behaviour is strange for me when I try to set the overboost
![image](https://user-images.githubusercontent.com/8193588/172006895-08f6295c-e64b-42cf-a063-cc4fc11ed462.png)
![image](https://user-images.githubusercontent.com/8193588/172006992-deb85735-9a5b-436f-aaa7-cb3c24a99e52.png)

**Comment by Pedro-Thales** [comment]
After this got stuck I closed the fan control and then opened the hwinfo and set the fans to manual it got unstuck, so set it to system auto, close the hwinfo and then opened the fan control again it was not stuck but the behaviour of setting overboost was weird as well
![image](https://user-images.githubusercontent.com/8193588/172008287-16c5fd76-46d3-4d2b-86cc-4702348f5056.png)

**Comment by T-Troll** [comment]
Hmm... Did you issue `setover` command then HWINFO/AlienFan working?
They contorol fans, so results wierd. Stop ALL fan-control software before check overboost!
But GPU fan stack is fascinating, how do they do it? Let me think and check...

Noted about G-Mode.

**Comment by Pedro-Thales** [comment]
Yes when I issued the setover the fancontrol was open.  I will try to set with everything closed

**Comment by T-Troll** [comment]
So, seems like it resolved. Feel free to re-open it in case still have some issues.

**Comment by Hellohi3654** [comment]
> After this got stuck I closed the fan control and then opened the hwinfo and set the fans to manual it got unstuck, so set it to system auto, close the hwinfo and then opened the fan control again it was not stuck but the behaviour of setting overboost was weird as well ![image](https://user-images.githubusercontent.com/8193588/172008287-16c5fd76-46d3-4d2b-86cc-4702348f5056.png)

I believe you have the same laptop as me or similar

first off try adding a sudden jump of fan boost at the start of your curves a bit like mine
![image](https://user-images.githubusercontent.com/57390764/173049723-d5d8a9c0-d3fe-49e9-8047-82cc9cf06c0f.png)

this seems to get rid of the issue where the fans will get stuck at low RPMs and then you get a BSOD

also, do
```
alienfan-cli setover=0,148
alienfan-cli setover=1,148
```
I found that our fans are able to go to 5700 on fan 0 (fan 1 in gui) and 5400 on fan 1 (fan 2 in gui) at least for me
I have a friend with the same laptop and his fans do the same RPMs as mine
btw if you use alienfan-gui's gmode and you have your overboost set to 148 your fans spin 700rpm faster than using AWCC gmode

personally I don't think hwinfo can effect your fans if you don't have it open
you can have hwinfo installed but don't make it start with windows

another thing I noticed is that if you have alienfan-gui set to start with windows and your not plugged into ac it generally gives a error and doesn't open but if you just go and manually open alienfan-gui it works fine

oh also avoid using alienfx-gui, alienfan-gui just seems to work better plus there's no point using alienfx-gui as we don't have any lights to actually use it with
only keyboard lights that have two modes

one last thing set your CPU boost on AC and battery to off 
if you don't care about losing 2-10 fps in most games it pretty much makes your CPU run 20C cooler which also keeps your GPU cooler which in some cases can actually increase fps
but if your just using your laptop for anything that's not CPU intense have it set to off, having it on aggressive only very slightly increases the loading speed of webpages and stuff and as a bonus your fans shouldn't ever turn on

when I have cpu boost set to off my CPU's temp is generally around 60-65C at idle but as soon as I switch to aggressive the temperature spikes to 75-85C and stays around that

**Comment by T-Troll** [comment]
> I believe you have the same laptop as me or similar

No, he has different model according to BIOS.

> btw if you use alienfan-gui's gmode and you have your overboost set to 148 your fans spin 700rpm faster than using AWCC gmode

This is the other side effect, boost doesn't affect it. I just set G-mode other way then AWCC for your hardware.

> oh also avoid using alienfx-gui, alienfan-gui just seems to work better plus there's no point using alienfx-gui as we don't have any lights to actually use it with only keyboard lights that have two modes

Wrong. Some g-series have the same 4-zone keyboard as Alienware. Also, fx-gui useful for profiles management (yes, fans included!). fan-gui is just a light-weighted version for people don't need this.

Another thing (i think about it last evening...) fx-gui currently noisy if no lights, and i will disable it next release.

> another thing I noticed is that if you have alienfan-gui set to start with windows and your not plugged into ac it generally gives a error and doesn't open but if you just go and manually open alienfan-gui it works fine

Ugh-oh. I'll check, interesting situation.

**Comment by T-Troll** [comment]
(offtopic)

> my CPU's temp is generally around 65-75C at idle

Repaste. colling system. It should be below 50.

**Comment by Hellohi3654** [comment]
> > I believe you have the same laptop as me or similar
> 
> No, he has different model according to BIOS.
> 
> > btw if you use alienfan-gui's gmode and you have your overboost set to 148 your fans spin 700rpm faster than using AWCC gmode
> 
> This is the other side effect, boost doesn't affect it. I just set G-mode other way then AWCC for your hardware.
> 
> > oh also avoid using alienfx-gui, alienfan-gui just seems to work better plus there's no point using alienfx-gui as we don't have any lights to actually use it with only keyboard lights that have two modes
> 
> Wrong. Some g-series have the same 4-zone keyboard as Alienware. Also, fx-gui useful for profiles management (yes, fans included!). fan-gui is just a light-weighted version for people don't need this.
> 
> Another thing (i think about it last evening...) fx-gui currently noisy if no lights, and i will disable it next release.
> 
> > another thing I noticed is that if you have alienfan-gui set to start with windows and your not plugged into ac it generally gives a error and doesn't open but if you just go and manually open alienfan-gui it works fine
> 
> Ugh-oh. I'll check, interesting situation.
> 
> PS: Why 5.9.4 CLI? Is something wrong with 6.1.1?

Oh I didn't know you could get 4 zone lighting 

I'd imagine the fans between most g series models would pretty much be the same
They all look very similar

Also the command line screenshot is because I quote replied to the message before 
6.1.1 cli is fine

**Comment by Hellohi3654** [comment]
> (off topic)
> 
> > my CPU's temp is generally around 65-75C at idle
> 
> Repaste cooling system. It should be below 50.

Yeah I was thinking about liquid metal but not sure if it would be worth the risk
Like is liquid metal another 10-20C cooler than paste?

**Comment by T-Troll** [comment]
>Oh I didn't know you could get 4 zone lighting

Yes, it's configuration variants.

>I'd imagine the fans between most g series models would pretty much be the same
> They all look very similar

But they have 3(!) different BIOS variants - i mean completely different!

So... I didn't reproduce fan-gui issue at batteries, my system start OK.
But for check, i'm add some delay for fan service start (this can be source of this issue, if system SO slow).
Can you please check this giu - [alienfan-gui.zip](https://github.com/T-Troll/alienfx-tools/files/8882870/alienfan-gui.zip)?
Please deselect and select back "Start with windows" to change some settings inside!
Also, this version should fix g-mode check box bug.

**Comment by Hellohi3654** [comment]
> >Oh I didn't know you could get 4 zone lighting
> 
> Yes, it's configuration variants.
> 
> >I'd imagine the fans between most g series models would pretty much be the same
> > They all look very similar
> 
> But they have 3(!) different BIOS variants - i mean completely different!
> 
> So... I didn't reproduce fan-gui issue at batteries, my system start OK.
> But for check, i'm add some delay for fan service start (this can be source of this issue, if system SO slow).
> Can you please check this giu - [alienfan-gui.zip](https://github.com/T-Troll/alienfx-tools/files/8882870/alienfan-gui.zip)?
> Please deselect and select back "Start with windows" to change some settings inside!
> Also, this version should fix g-mode check box bug.
> 

I tried adding 10 seconds of delay to a task scheduler task but that didn't fix the issue for at least 
It just seems like it needs to be opened twice for some reason
Maybe you could add a 10 second timer to the not supported hardware popup that tries to reopen the app after 10 seconds if not interrupted by clicking cancel
Also the bug is a little annoying because it disables the start with windows option. Could you just add the disable start with windows option to the not supported hardware popup?
Thanks

I will try the new gui to see if it fixed the gmode button bug

**Comment by Hellohi3654** [comment]
yes that new gui fixes the gmode button bug
It seems to be working fine now

**Comment by T-Troll** [comment]
> I tried adding 10 seconds of delay to a task scheduler task but that didn't fix the issue for at least
> It just seems like it needs to be opened twice for some reason

No, second start didn't change anything.
Can you please check what working path set into scheduler? THIS can fail, so it can't locate driver/loader!

**Comment by Hellohi3654** [comment]
> > I tried adding 10 seconds of delay to a task scheduler task but that didn't fix the issue for at least
> > It just seems like it needs to be opened twice for some reason
> 
> No, second start didn't change anything. Can you please check what working path set into scheduler? THIS can fail, so it can't locate driver/loader!

I just copied all the settings from your task scheduler task and then just added this
![image](https://user-images.githubusercontent.com/57390764/173168318-a65886c9-b64b-4afc-aff4-9b5cdd6470bd.png)

**Comment by Hellohi3654** [comment]
> > > I tried adding 10 seconds of delay to a task scheduler task but that didn't fix the issue for at least
> > > It just seems like it needs to be opened twice for some reason
> > 
> > 
> > No, second start didn't change anything. Can you please check what working path set into scheduler? THIS can fail, so it can't locate driver/loader!
> 
> I just copied all the settings from your task scheduler task and then just added this ![image](https://user-images.githubusercontent.com/57390764/173168318-a65886c9-b64b-4afc-aff4-9b5cdd6470bd.png)

I'm pretty sure I actually just exported your task and then reimported it

**Comment by T-Troll** [comment]
Heh! "Delay" there is broken and didn't work. I use other way to delay start.
So... Can you run it from task scheduler MANUALLY after you have desktop? If yes - some service can't start yet. If no - issue is in task settings.

**Comment by T-Troll** [comment]
Upd: try this, it uses notification toasts instead of message boxes. 
[alienfan-gui.zip](https://github.com/T-Troll/alienfx-tools/files/8882968/alienfan-gui.zip)
Also, delay increased.

**Comment by Hellohi3654** [comment]
oh just found out the problem is worst than I thought

![image](https://user-images.githubusercontent.com/57390764/173171123-4cd7c30e-35b2-4a75-9b92-ba94507f2520.png)

Alienfan-gui won't start at all on battery just keeps showing this error

**Comment by Hellohi3654** [comment]
![image](https://user-images.githubusercontent.com/57390764/173171170-46989629-3666-44cc-a88e-08ce78f0ee30.png)

**Comment by T-Troll** [comment]
This error message means it can't load the driver.
Why on battery? Some changes in system security settings? Maybe now it power-related in W11?
Check files (KDL, hwacc) still in place and system security the same!

**Comment by Hellohi3654** [comment]
ok the problem is now extremely bad I cannot get fan control to work at all
even when plugged into AC
not versions work nothing
tried restarting 
force restarts 
nothing is working

**Comment by T-Troll** [comment]
Check recent W11 updates and settings changes.
It's a menace....

**Comment by Hellohi3654** [comment]
no there was no update and my antivirus hasn't changed or anything

**Comment by Hellohi3654** [comment]
I think it's possible that the driver is broken
how do you unload it?

**Comment by Hellohi3654** [comment]
C:\Alienfan\KDU-master\KDU-master\Bin>kdu.exe -prv 1 -scv 3 -drvn HwAcc -map HwAcc.sys
[#] Kernel Driver Utility v1.2.0 (build 2202) started, (c)2020 - 2022 KDU Project
[#] Build at Thu Feb 17 01:33:48 2022, header checksum 0x4D588
[#] Supported x64 OS : Windows 7 and above
[*] Windows version: 10.0 build 22621
[*] SecureBoot is enabled on this machine
[+] Selected provider: 1
[*] Driver mapping using shellcode version: 3
[+] Driver object name: "HwAcc"
[+] No driver registry key name specified, driver object name will be used instead
[+] Input driver file "HwAcc.sys" loaded at 0x00007FF774650000
[+] Provider: "CVE-2019-16098", Name "RTCore64"
[+] Drivers database "drv64.dll" loaded at 0x000002031ADC0002
[!] Vulnerable driver is already loaded
[!] Unable to open vulnerable driver, NTSTATUS (0xC000000E)
[+] Return value: 0. Bye-bye!

C:\Alienfan\KDU-master\KDU-master\Bin>

**Comment by Hellohi3654** [comment]
it says it's loaded but every version of alienfan-gui doesn't open

**Comment by Hellohi3654** [comment]
C:\Alienfan\KDU-master\KDU-master\Bin>kdu.exe -prv 1 -scv 3 -drvn HwAcc -map HwAcc.sys
[#] Kernel Driver Utility v1.2.0 (build 2202) started, (c)2020 - 2022 KDU Project
[#] Build at Thu Feb 17 01:33:48 2022, header checksum 0x4D588
[#] Supported x64 OS : Windows 7 and above
[*] Windows version: 10.0 build 22621
[*] SecureBoot is enabled on this machine
[+] Selected provider: 1
[*] Driver mapping using shellcode version: 3
[+] Driver object name: "HwAcc"
[+] No driver registry key name specified, driver object name will be used instead
[+] Input driver file "HwAcc.sys" loaded at 0x00007FF601080000
[+] Provider: "CVE-2019-16098", Name "RTCore64"
[+] Drivers database "drv64.dll" loaded at 0x000002391EC50002
[!] Vulnerable driver is already loaded
[+] Vulnerable driver "RTCore64" opened
[+] Executing post-open callback for given provider
[+] Driver device security descriptor set successfully
[+] Victim "PROCEXP152" 1 acquire attempt of 3 (max)
[+] Processing victim "Process Explorer" driver
[+] Extracting victim driver "PROCEXP152" as "C:\WINDOWS\system32\drivers\PROCEXP152.sys"
[+] Victim is accepted, handle 0x00000000000000A4
[+] Reading FILE_OBJECT at 0xFFFFC508A4428B30
[!] Could not read FILE_OBJECT at 0xFFFFC508A4428B30
[!] Error preloading victim driver, abort
[+] Victim released
[+] Vulnerable driver "RTCore64" unloaded
[+] Return value: 0. Bye-bye!

C:\Alienfan\KDU-master\KDU-master\Bin>

**Comment by Hellohi3654** [comment]
```
C:\Alienfan\KDU-master\KDU-master\Bin>kdu.exe -prv 20 -scv 3 -drvn HwAcc -map HwAcc.sys
[#] Kernel Driver Utility v1.2.0 (build 2202) started, (c)2020 - 2022 KDU Project
[#] Build at Thu Feb 17 01:33:48 2022, header checksum 0x4D588
[#] Supported x64 OS : Windows 7 and above
[*] Windows version: 10.0 build 22621
[*] SecureBoot is enabled on this machine
[+] Selected provider: 20
[*] Driver mapping using shellcode version: 3
[+] Driver object name: "HwAcc"
[+] No driver registry key name specified, driver object name will be used instead
[+] Input driver file "HwAcc.sys" loaded at 0x00007FF738C70000
[+] Provider: "CVE-2021-36276", Name "DBUtilDrv2"
[+] Drivers database "drv64.dll" loaded at 0x000002003CD00002
[!] Vulnerable driver is already loaded
[+] Vulnerable driver "DBUtilDrv2" opened
[+] Executing post-open callback for given provider
[+] Victim "PROCEXP152" 1 acquire attempt of 3 (max)
[+] Processing victim "Process Explorer" driver
[+] Extracting victim driver "PROCEXP152" as "C:\WINDOWS\system32\drivers\PROCEXP152.sys"
[+] Victim is accepted, handle 0x00000000000001AC
[+] Reading FILE_OBJECT at 0xFFFF9C81E966E530
[+] Reading DEVICE_OBJECT at 0xFFFF9C81F1EDFE10
[+] Reading DRIVER_OBJECT at 0xFFFF9C81F28C09F0
[+] Victim IRP_MJ_DEVICE_CONTROL 0xFFFFF8049B952220
[+] Victim DriverUnload 0xFFFFF8049B953280
[+] Loaded ntoskrnl base 0xFFFFF80436E00000
[+] Ntoskrnl.exe mapped at 0x7FF6081C0000
[+] Resolving kernel import for input driver
[+] Resolving payload import
[*] IoDriverObjectType 0xFFFFF80437B1DBB0
[*] ObCreateObject 0xFFFFF80437589D20
[*] ObInsertObject 0xFFFFF804374ADC20
[*] ObMakeTemporaryObject 0xFFFFF804375D7C00
[*] ZwMakeTemporaryObject 0xFFFFF8043721B640
[*] ZwClose 0xFFFFF80437219540
[+] Resolving base shellcode import
[*] MmSectionObjectType 0xFFFFF80437B1DB28
[*] ExAllocatePoolWithTag 0xFFFFF80437898CC0
[*] ExFreePoolWithTag 0xFFFFF80437898010
[*] IofCompleteRequest 0xFFFFF8043709B400
[*] ZwMapViewOfSection 0xFFFFF80437219860
[*] ZwUnmapViewOfSection 0xFFFFF804372198A0
[*] ObReferenceObjectByHandle 0xFFFFF804374C86D0
[*] ObfDereferenceObject 0xFFFFF80437093990
[*] KeSetEvent 0xFFFFF80437088680
[+] Bootstrap code size = 0x4E1
[+] Driver IRP_MJ_DEVICE_CONTROL handler code modified
[+] Run shellcode
[~] Shellcode result: NTSTATUS (0x0)
[+] Victim released
[+] Return value: 1. Bye-bye!
```

Found one that works

**Comment by Hellohi3654** [comment]
Seems like the other one is now patched

**Comment by Hellohi3654** [comment]
Found possible reason why the other one is patched
![image](https://user-images.githubusercontent.com/57390764/173207782-cdc5236a-a1db-4c43-84a8-42c27c9732f7.png)

**Comment by Hellohi3654** [comment]
I'm not even in the dev channel anymore so it was probably patched for a little while
![image](https://user-images.githubusercontent.com/57390764/173207814-de5d154f-e2ac-4cac-b2d8-48f2636b6535.png)

**Comment by T-Troll** [comment]
Seems like it's other. Then using RTCore you have this message in KDU log:

```
[+] Victim "PROCEXP152" 1 acquire attempt of 3 (max)
[+] Processing victim "Process Explorer" driver
[+] Extracting victim driver "PROCEXP152" as "C:\WINDOWS\system32\drivers\PROCEXP152.sys"
[+] Victim is accepted, handle 0x00000000000000A4
[+] Reading FILE_OBJECT at 0xFFFFC508A4428B30
[!] Could not read FILE_OBJECT at 0xFFFFC508A4428B30
[!] Error preloading victim driver, abort
```
Seems like system block access to system32!

Let me try to unpack and load it from app folder instead....

**Comment by T-Troll** [comment]
Ok, let's try to use this KDL - [kdl.zip](https://github.com/T-Troll/alienfx-tools/files/8885262/kdl.zip)

**Comment by T-Troll** [comment]
This should not be like this, even at W11.
Something block read access to driver file, in fact.

Hmm... Can you try other app path (something simple from C: root)? Here can be an issue.

**Comment by ash-hashtag** [comment]
I have the same device and same problem

**Comment by T-Troll** [comment]
Problem is not device, but f***ing Windows 11. Every update, they eat my brain!
I'm starting to think not support it officially...

You can try @Hellohi3654 method described above meanwhile.

**Comment by T-Troll** [comment]
No, but you can port my SDKs. VM will not work at all (low-level hardware access requires).

**Comment by Pedro-Thales** [comment]
I have a dell 5515 ryzen edition(bios 1.6.0) running windows 11 education and the version 6.2.0 is working fine.

I'm using only the fan-gui and monitor

**Comment by T-Troll** [comment]
It's something into Windows 11 security settings or Defender configuration (maybe beta/insider channel as well).
We are trying to pinpoint it, but still have issues from time to time.

**Comment by Hellohi3654** [comment]
> Found possible reason's why the other one is patched
> ![image](https://user-images.githubusercontent.com/57390764/173207782-cdc5236a-a1db-4c43-84a8-42c27c9732f7.png)
> 

I'm pretty sure it's one of these updates that done it
For me that is the only thing that could of done it as there was no windows update between it working and breaking and I was using the same version in the same location with the same settings when it broke

**Comment by Hellohi3654** [comment]
> I have a dell 5515 ryzen edition(bios 1.6.0) running windows 11 education and the version 6.2.0 is working fine.
> 
> I'm using only the fan-gui and monitor

I think it's only insider builds it's currently not working for

**Comment by Hellohi3654** [comment]
> This should not be like this, even at W11.
> Something block read access to driver file, in fact.
> 
> Hmm... Can you try other app path (something simple from C: root)? Here can be an issue.

I haven't tried that yet but I don't think it will fix it tbh

**Comment by T-Troll** [comment]
According to your logs, the issue is something to block the driver (leading one, not mine) file from loading.
So, what happened? KDL/KDU unpack leading driver to disk, but can't load it - file object not found. So - it can be a bug into path (W11-specific path processing) or Defender block (it can't be disabled completely from settings - some components continue working, but by registry/policy only).

In latest KDLs i change path to app folder, but, as you can see, KDU use Windows folder instead (with the same result).

**Comment by T-Troll** [comment]
Closed as duplicate of #114, gear-related issue is fixed.
